src/sparse/impl/KokkosSparse_spgemm_impl_triangle.hpp

```diff
@@ -1123,21 +1111,23 @@ struct KokkosSPGEMM
   void operator()(const MultiCoreTag2&, const team_member_t & teamMember) const {
     const nnz_lno_t team_row_begin = teamMember.league_rank() * team_row_chunk_size;
     const nnz_lno_t team_row_end = KOKKOSKERNELS_MACRO_MIN(team_row_begin + team_row_chunk_size, numrows);
-
+    const nnz_lno_t chunk_size = 0;
 
     //get memory from memory pool.
     volatile nnz_lno_t * tmp = NULL;
     nnz_lno_t tid = get_thread_id(team_row_begin + teamMember.team_rank());
     while (tmp == NULL){
       tmp = (volatile nnz_lno_t * )( m_space.allocate_chunk(tid));
+      // issue-508, TODO: chunk_size = ??
     }
 
     //set first to globally used hash indices.
     nnz_lno_t *globally_used_hash_indices = (nnz_lno_t *) tmp;
     tmp += pow2_hash_size;
 
     //create hashmap accumulator.
-    KokkosKernels::Experimental::HashmapAccumulator<nnz_lno_t,nnz_lno_t,nnz_lno_t> hm2;
+    KokkosKernels::Experimental::HashmapAccumulator<nnz_lno_t,nnz_lno_t,nnz_lno_t,KokkosKernels::Experimental::HashOpType::bitwiseAnd> 
+    hm2(chunk_size, pow2_hash_func, nullptr, nullptr, nullptr, nullptr);
```

Comment:
Same question as above about chunk_size=0

Reply:
Good catch! chunk_size should be MaxRoughNonZero (see https://github.com/kokkos/kokkos-kernels/pull/731/files/ebab3bc86c0e9aebf28b0cc3cdaaaa96419bf058#diff-12c0beaaa61cf811b978be6f4dc44d77L1160)!

Reply:
Same as https://github.com/kokkos/kokkos-kernels/pull/731#discussion_r440438891.